src/rviz/visualization_frame.cpp

```diff
@@ -1185,6 +1185,17 @@ void VisualizationFrame::addTool(Tool* tool)
   tool_to_action_map_[tool] = action;
 
   remove_tool_menu_->addAction(tool->getName());
+
+  QObject::connect(tool, &Tool::nameChanged, this, &VisualizationFrame::updateTool);
+}
+
+void VisualizationFrame::updateTool(Tool* tool)
+{
+  // Early return if the tool is not present
+  if (tool_to_action_map_.find(tool) == tool_to_action_map_.end())
+    return;
+  QAction* action = tool_to_action_map_[tool];
+  action->setIconText(tool->getName());
```

Comment:
Please avoid searching the map twice:
```suggestion
  auto it = tool_to_action_map_.find(sender());
  if (it == tool_to_action_map_.end())
    return;
  (*it)->setIconText(name);
```